### PR TITLE
Match Func i1 801922 dc

### PR DESCRIPTION
**Pull request by Vulto**

### CODE REVIEW

File: src/overlays/ovl_i1/sf_ve1.c [hunk 1]
```diff
@@ -1,13 +1,123 @@
...
+            arg0->obj.status = 0;
+            return;
+        }
+    } else {
+        var_v0 = (Boss*) ((u32) var_v0 + 0x408);
+        if (var_v1 != 4) {
+            goto loop_1;
+        }
+    }
+}
```
Comment (sonicdcer) [inline src/overlays/ovl_i1/sf_ve1.c:120]:
```suggestion
void func_i1_801923C0(Object_4C* obj4C) {
    Boss* boss;
    s32 i;

    for (boss = &gBosses[0], i = 0; i < 4; i++, boss++) {
        if (boss->obj.status != 0 && boss->obj.id == OBJ_BOSS_319) {
            if (boss->obj.pos.z <= obj4C->obj.pos.z) {
                D_i1_8019C0BC = (s32)obj4C->obj.rot.x + 1;
                obj4C->obj.status = 0;
            }
            break;
        }
    }
}

**Review by sonicdcer — CHANGES_REQUESTED** [review]

File: src/overlays/ovl_i1/sf_ve1.c (hunk 1 above)
Comment (sonicdcer) [inline src/overlays/ovl_i1/sf_ve1.c:120]:
```suggestion
void func_i1_801923C0(Object_4C* obj4C) {
    Boss* boss;
    s32 i;

    for (boss = &gBosses[0], i = 0; i < 4; i++, boss++) {
        if (boss->obj.status != 0 && boss->obj.id == OBJ_BOSS_319) {
            if (boss->obj.pos.z <= obj4C->obj.pos.z) {
                D_i1_8019C0BC = (s32)obj4C->obj.rot.x + 1;
                obj4C->obj.status = 0;
            }
            break;
        }
    }
}

**Review by sonicdcer — CHANGES_REQUESTED** [review]

File: src/overlays/ovl_i1/sf_ve1.c (hunk 1 above)
Comment (Vulto) [reply src/overlays/ovl_i1/sf_ve1.c:120]: Much better now. TY

File: src/overlays/ovl_i1/sf_ve1.c [hunk 2]
```diff
@@ -1,13 +1,118 @@
...
+    temp_fv1_3 = *arg0;
+
+    if (temp_fv1_3 >= 360.0f) {
+        *arg0 = temp_fv1_3 - 360.0f;
+    } else if (temp_fv1_3 < 0.0f) {
+        *arg0 = temp_fv1_3 + 360.0f;
+    }
+
+    return var_fa1;
+}
```
Comment (sonicdcer) [inline src/overlays/ovl_i1/sf_ve1.c:70]: https://decomp.me/scratch/EiXRu
```suggestion
f32 func_i1_801920F0(f32* arg0, f32 arg1, f32 arg2, f32 arg3, f32 arg4, f32* arg5) {
    f32 temp;

    *arg5 = arg1 - *arg0;

    if (*arg5 > 180.0f) {
        do {
            *arg5 -= 360.0f;
        } while (*arg5 > 180.0f);
    } else if (*arg5 < -180.0f) {
        do {
            *arg5 += 360.0f;
        } while (*arg5 < -180.0f);
    }

    temp = *arg5;

    if (temp != 0.0f) {
        temp *= arg2;

        if ((arg4 <= temp) || ((temp <= -arg4))) {
            if (arg3 < temp) {
                temp = arg3;
            } else {

                if (-arg3 > temp) {
                    temp = -arg3;
                }
            }
            *arg0 += temp;
        } else {
            if (temp < arg4) {
                temp = arg4;
                *arg0 += arg4;

                if (arg1 < *arg0) {
                    *arg0 = arg1;
                }
            } else if (-arg4 < temp) {
                temp = -arg4;
                *arg0 += -arg4;

                if (*arg0 < arg1) {
                    *arg0 = arg1;
                }
            }
        }
    }

    if (*arg0 >= 360.0f) {
        *arg0 = *arg0 - 360.0f;
    } else if (*arg0 < 0.0f) {
        *arg0 = *arg0 + 360.0f;
    }

    return temp;
}

File: src/overlays/ovl_i1/sf_ve1.c [hunk 3]
```diff
@@ -1,13 +1,118 @@
...
 
-#pragma GLOBAL_ASM("asm/us/nonmatchings/overlays/ovl_i1/sf_ve1/func_i1_801922DC.s")
+void func_i1_801922DC(Object_4C* this) {
+    Boss* boss = gBosses;
```
Comment (sonicdcer) [inline src/overlays/ovl_i1/sf_ve1.c:73]:
```suggestion
Boss* boss = &gBosses[0];

Comment (sonicdcer) [reply src/overlays/ovl_i1/sf_ve1.c:73]: same for the other ones

**Review by sonicdcer — CHANGES_REQUESTED** [review]

File: src/overlays/ovl_i1/sf_ve1.c [hunk 4]
```diff
@@ -1,15 +1,133 @@
...
+            break;
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/us/nonmatchings/overlays/ovl_i1/sf_ve1/func_i1_8019243C.s")
+void func_i1_801923C0(Object_4C* obj4C) {
+    Boss* boss;
+    s32 i;
+    for (boss = &gBosses[0], i = 0; i < 4; i++, boss++) {
```
Comment (sonicdcer) [inline src/overlays/ovl_i1/sf_ve1.c:106]: Space between declarations and code, full Style guide here:
https://github.com/sonicdcer/sf64/wiki/Style-guide

File: src/overlays/ovl_i1/sf_ve1.c [hunk 5]
```diff
@@ -1,15 +1,133 @@
...
+                this->obj.status = 0;
+            }
+            break;
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/us/nonmatchings/overlays/ovl_i1/sf_ve1/func_i1_80192358.s")
+void func_i1_80192358(Object_4C* this) {
+    Boss* boss = gBosses;
```
Comment (sonicdcer) [inline src/overlays/ovl_i1/sf_ve1.c:89]: There are still others to fix, please don't use the apply suggestion button, you should do this manually.
```suggestion
Boss* boss = &gBosses[0];

**Review by sonicdcer — CHANGES_REQUESTED** [review]

**Review by sonicdcer — APPROVED** [review]

**Pull request merged**